### PR TITLE
Diff module (source + tests)

### PR DESCRIPTION
**Pull request by SatyendraBanjare**
Added lua bindings for the functions of diff module.

### CODE REVIEW

File: tests/diff/test_diff_spec.lua [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    local repo_path = "Fixtures/WORKON_REPO"
+    local diff_options, repo_index
+
+    setup(function()
+        luagit2.init()
+    end)
+    
+    before_each(function()
+        fixer.set_repo("new_test_repo")
+        repo = luagit2.repository_open(repo_path)
```
Comment (Etiene) [inline tests/diff/test_diff_spec.lua:16]: `repo` is global here :) you can fix it in this PR, but I suggest that once all PRs are merged, that you run luacheck on the whole project just to see if there is anything else we didnt catch :)

Comment (SatyendraBanjare) [reply tests/diff/test_diff_spec.lua:16]: yes mam, surely :smile:

File: tests/diff/test_diff_spec.lua [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        local diff_data_name_only = luagit2.buf_details(diff_idx_wkdir_buf_name_only)
+
+        local diff_idx_wkdir_buf_name_status = luagit2.diff_to_buf(diff_idx_wkdir,diff_format_options_name_status)
+        local diff_data_name_status = luagit2.buf_details(diff_idx_wkdir_buf_name_status)
+
+        print("diff Raw : \n\n" .. diff_data_raw .. "\n\n")
+        print("diff patch: \n\n" .. diff_data_patch .. "\n\n")
+        print("diff patch header: \n\n" .. diff_data_patch_header .. "\n\n")
+        print("diff name only : \n\n" .. diff_data_name_only .. "\n\n")
+        print("diff name status : \n\n" .. diff_data_name_status .. "\n\n")
```
Comment (Etiene) [inline tests/diff/test_diff_spec.lua:77]: Why are there print statements instead of asserts here?

File: tests/diff/test_diff_spec.lua [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        
+        local stats_short_buf = luagit2.diff_stats_to_buf(diff_stats,stats_format_short,20)
+        local data_stats_short = luagit2.buf_details(stats_short_buf)
+
+        local stats_number_buf = luagit2.diff_stats_to_buf(diff_stats,stats_format_number,20)
+        local data_stats_number = luagit2.buf_details(stats_number_buf)
+
+        print("diff stats full: \n\n" .. data_stats_full .. "\n\n")
+        print("diff stats short: \n\n" .. data_stats_short .. "\n\n")
+        print("diff patch number: \n\n" .. data_stats_number .. "\n\n")
```
Comment (Etiene) [inline tests/diff/test_diff_spec.lua:113]: Same question as above

**Pull request merged**